test_pull_requests

```diff
@@ -480,79 +712,80 @@ module Hub
         end
-        commit = get_last_commit(pull_request['number'], repo)
-        update_status(settings['test_prefix'], commit['sha'], repo, 'pending', build_url, 'Testing')
       end
-
       submitted_tests[settings['name']] = true unless settings['allow_multiple']
     end
 
     def mark_test_success(pull_id, repo, settings)
       $stderr.puts "  Marking SUCCESS for pull request ##{pull_id} in repo '#{repo}'"
       bot_comment = get_comment_with_prefix(pull_id, repo, settings['test_prefix'])
-      build_url = nil
-      extended_tests = nil
-      if bot_comment['body'] =~ /^#{settings['test_prefix']} Running \(([^\)]+)\)( \(Extended Tests: [^\)]+\).*)?/
-        build_url = $1
-        extended_tests = $2
-      else
-        raise "'#{settings['name']}' pull request not found"
-      end
-      comment = "#{settings['test_prefix']} SUCCESS (#{build_url})"
-      comment += extended_tests if extended_tests
+      fields = extract_bot_comment_fields(bot_comment['body'], settings)
+      fields[:state] = :success
+      comment = compose_bot_comment(settings['test_prefix'], fields)
       recreate_comment(pull_id, bot_comment['id'], repo, comment)
 
-      commit = get_last_commit(pull_id, repo)
-      update_status(settings['test_prefix'], commit['sha'], repo, 'success', build_url, 'Passed')
+      commit = last_commit_for_pull_id(pull_id, repo)
+      update_status(settings['test_prefix'], commit['sha'], repo, 'success', fields[:build_url], 'Passed')
     end
 
-    def merge_pull_request(pull_id, repo, settings, merge_pretest_success_url=nil)
+    def merge_pull_request(pull_id, repo, settings, merge_pretest_success_url=nil, fields=nil)
       $stderr.puts "  Merging pull request ##{pull_id} for repo '#{repo}'"
       pull_request, bot_comment, comments = test_merge_pull_request(pull_id, repo, settings)
-      build_url = nil
+      if fields.nil?
+        fields = extract_bot_comment_fields(bot_comment['body'], settings)
+      end
+      fields[:state] = :success
       if merge_pretest_success_url
+        fields[:build_url] = merge_pretest_success_url
         build_url = merge_pretest_success_url
-      elsif bot_comment['body'] =~ /^#{settings['test_prefix']} Running \((.*)\)/
-        build_url = $1
-      else
-        raise "'#{settings['name']}' pull request not found"
       end
-      comment = "#{settings['test_prefix']} SUCCESS (#{build_url})"
+      comment = compose_bot_comment(settings['test_prefix'], fields)
+
       begin
         branch = pull_request['base']['ref']
         b_settings = branch_settings(branch, settings)
         image_base_name = b_settings['image_base_name']
         if image_base_name
           image_base_name.gsub!('*', branch)
           image = "#{image_base_name}_#{JenkinsAPI.get_next_build(branch, settings, b_settings['downstream_job_name'])}"
+          # TODO: Add this to compose_bot_comment
```

Comment:
But I think this particular change isn't worth complicating this PR right now, so I deferred it